core-bundle/src/DependencyInjection/Configuration.php

```diff
@@ -901,4 +903,31 @@ static function (array $allowedProperties): array {
             ->end()
         ;
     }
+
+    private function addAltchaNode(): NodeDefinition
+    {
+        return (new TreeBuilder('altcha'))
+            ->getRootNode()
+            ->addDefaultsIfNotSet()
+            ->children()
+                ->enumNode('algorithm')
+                    ->info('The algorithm used to generate the challenges. Select between "SHA-256", "SHA-384" or "SHA-512".')
+                    ->values(Algorithm::values())
+                    ->defaultValue(Algorithm::sha256->value)
+                ->end()
+                ->integerNode('range_min')
+                    ->info('A higher value increases the complexity/security but may significantly increase the computational load on client devices, potentially impacting the user experience.')
+                    ->defaultValue(10000)
+                ->end()
+                ->integerNode('range_max')
+                    ->info('A higher value increases the complexity/security but may significantly increase the computational load on client devices, potentially impacting the user experience.')
+                    ->defaultValue(100000)
+                ->end()
+                ->integerNode('challenge_expiry')
+                    ->info('The time period in seconds for which a challenge is valid.')
```

Comment:
Imho we should force a `->min()` of `3600` for security reasons. We can still lower the value if anybody submits an issue with a reasonable argument why doing so makes sense.

Reply:
Aren‘t shorter periods more secure?

Reply:
Well, if it expires quickly and the cronjob cleans them up fast, you have basically no protection?

Reply:
But the parameter also controls how long a challenge is valid. So maybe we should have different parameters for how long a challenge is valid and how long a solved challenge remains in the DB to mitigate replay attacks?

Reply:
Ah, I see. Probably yes although I don't know if we really need to have the replay attack mitigation period configurable. Wdyt @ausi ?

Reply:
I think we can simply keep the DB entries for 24 hours and set the `challenge_expiry` to a short period like 10 minutes.

Reply:
> So maybe we should have different parameters for how long a challenge is valid and how long a solved challenge remains in the DB to mitigate replay attacks?

I think they have to be the same. As long as a challenge is valid, it has to stay in the DB to mitigate replay attacks. After a challenge is expired there is no need to store it in DB anymore as it would always be invalid.

Reply:
Changed in 2d906c11799622aaa61fcb52dcccc38380b92e38.